modules/library/main/src/main/java/org/geotools/styling/visitor/RescaleStyleVisitor.java

```diff
@@ -133,6 +133,34 @@ protected Expression rescale( Expression expr ) {
     }
 
     /**
+     * Rescale using listMultiply, if there is only one entry.
+     *
+     * @param expressions
+     * @return
+     */
+    protected List<Expression> rescaleDashArray(List<Expression> expressions) {
+        if (expressions == null || expressions.isEmpty()) {
+            return expressions;
+        }
+        
+        Expression rescaleToExpression = rescale(ff.literal(1));
+        //How to test, if it is a measure with a unit or not?
+        String data = ((Literal)rescaleToExpression).getValue().toString();
+        boolean evaluate = rescaleToExpression instanceof Literal && Character.isDigit(data.charAt(data.length()-1));
+        
+        List<Expression> rescaled = new ArrayList<>(expressions.size());
+        for (Expression expression : expressions) {
+            Expression rescale = ff.function("listMultiply", rescaleToExpression, expression);
+            if (expression instanceof Literal && evaluate) {
+                rescaled.add(ff.literal(rescale.evaluate(null)));
+            } else {
+                rescaled.add(rescale);
+            }
+        }
+        return rescaled;
```

Comment:
My intention was to precalculate all possible stuff. Otherwise one has to calculate it for every feature that hits this rule.